src/main/java/Deadline.java

```diff
@@ -4,7 +4,11 @@
 
     public Deadline(String description, String by) {
         super(description);
-        this.by = by;
+        this.by = by.trim();
+    }
+
+    public boolean checkDateline() {
```

Comment:
just another small coding standard violation, names representing methods should follow camelCase style, (i.e should the method should be named "checkDateLine" instead of "checkDateline"